src/panels/config/lovelace/resources/dialog-lovelace-resource-detail.ts

```diff
@@ -23,6 +23,18 @@ import { haStyleDialog } from "../../../../resources/styles";
 import { HomeAssistant } from "../../../../types";
 import { LovelaceResourceDetailsDialogParams } from "./show-dialog-lovelace-resource-detail";
 
+const styleSheetExt = ["css", "scss", "sass"];
+
+const detectResourceType = (url: string) => {
+  const ext = url.split(".").pop() || "";
+
+  if (styleSheetExt.includes(ext)) {
+    return "css";
+  }
+
+  return "module";
+};
```

Comment:
We should only automatically set the type when the extension is `.js` or `.css` and only if the type was not set before by the user. So we should then start with an empty value for type.